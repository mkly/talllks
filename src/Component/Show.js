import { html } from "htm/preact";
import { css } from "emotion";

const buttonStyle = css`
  padding: 25px;
  width: 100%;
  margin: 0 21px 21px 21px;
  max-width: 420px;
  background: orange;
  border-radius: 4px;
  border: 0;
  box-shadow: 1px 1px 2px 0 rgba(0, 0, 0, 0.3);
  font-size: 22px;
  font-family: monospace;
  color: white;
  font-weight: bold;
  letter-spacing: 2px;
  text-transform: uppercase;
  background: #222;
  background: linear-gradient(to top, #313131, #505050);
  &:hover,
  &:active {
    cursor: pointer;
  }
  &:active {
    box-shadow: none;
  }
  &:focus {
    outline: 0;
  }
`;

const onClick = callback => callback;

const Show = ({ showCowHello, text }) => html`
  <button onClick=${onClick(showCowHello)} class=${buttonStyle}>
    ${text}
  </button>
`;

export default Show;
