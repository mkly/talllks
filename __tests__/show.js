/**
 * @jest-environment jsdom
 */
import { html, render } from "htm/preact";

import Show from "../src/Component/Show";

describe("Show", () => {
  it("Should show the button", () => {
    const body = document.createElement("div");
    document.body.appendChild(body);
    render(html`<${Show} text="Testing 123"/>`, body);
    expect(body.innerHTML).toMatchSnapshot();
  });
});
