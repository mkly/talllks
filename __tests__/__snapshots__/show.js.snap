// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Should show the button 1`] = `"<div class=\\"css-1qqjzai\\"><button class=\\"css-1qfxjdt\\">Testing 123</button></div>"`;
